corehq/apps/domain/models.py

```diff
@@ -788,6 +788,9 @@ def copy_component(self, doc_type, id, new_domain_name, user=None):
         if self.is_snapshot and doc_type == 'Application':
             new_doc.prepare_multimedia_for_exchange()
 
+        # when copying from app-docs that don't have
+        # unique_id attribute on Modules
+        new_doc.ensure_module_unique_ids(should_save=False)
```

Comment:
ha! Should be good if I put this inside conditional check bloc for `Application` or `RemoteApp`.